quickstarts/microsoft.datafactory/data-factory-v2-azure-sql-database-to-sql-data-warehouse-copy/prereqs/prereq.azuredeploy.json

```diff
@@ -110,25 +115,28 @@
           "properties": {
             "collation": "[variables('dataWarehouseCollation')]",
             "edition": "[variables('dataWarehouseEdition')]",
-            "maxSizeBytes": "263882790666240",
             "requestedServiceObjectiveName": "[variables('dataWarehouseServiceObjectiveName')]"
           }
         }
       ]
     }
   ],
   "outputs": {
-    "storageCS": {
+    "storageAccountName": {
+      "type": "string",
+      "value": "[variables('storageAccountName')]"
+    },
+    "storageAccountResourceId": {
       "type": "string",
-      "value": "[concat('DefaultEndpointsProtocol=https;AccountName=',variables('storageAccountName'),';AccountKey=',listKeys(resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName')), '2016-01-01').keys[0].value)]"
+      "value": "[resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName'))]"
     },
     "sqlDatabaseCS": {
       "type": "string",
-      "value": "[concat('Server=tcp:', variables('serverName'),'.database.windows.net,1433;Initial Catalog=',variables('databaseName'), ';Persist Security Info=False;User ID=', parameters('databaseAdminUsername'), ';Password=', parameters('databaseAdminPassword'), ';Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;')]"
+      "value": "[concat('Server=tcp:', variables('serverName'), parameters('sqlServerBase'), ',1433;Initial Catalog=',variables('databaseName'), ';Persist Security Info=False;User ID=', parameters('databaseAdminUsername'))]"
```

Comment:
this is a really strange pattern - we're outputting part of a connectionString - which is usually a secret.  So it looks like we have a parameter in the template which is a secret but is not a secureString.  I think we have a few options
1) rename this output (and the param) to make it clear it's not a secret or 
1) construct then entire connstr in the main template (by passing in all the parts needed to construct it
1) make it a secret and store it in keyVault (and then retrieve it from there in the template

Reply:
Updated with option 2